browser/ui/views/speedreader/speedreader_mode_bubble.cc

```diff
@@ -124,9 +126,8 @@ void SpeedreaderModeBubble::Init() {
   // float button right
   site_toggle_layout->set_main_axis_alignment(
       views::BoxLayout::MainAxisAlignment::kEnd);
-  auto site_toggle_button =
-      std::make_unique<views::ToggleButton>(base::BindRepeating(
-          &SpeedreaderModeBubble::OnButtonPressed, base::Unretained(this)));
+  auto site_toggle_button = std::make_unique<views::ToggleButton>();
+  site_toggle_button->SetIsOn(tab_helper_->IsEnabledForSite());
```

Comment:
if I toggle off speedreader globally, it shows a disabled toggle (on a distilled page), not sure whether it is expected

Reply:
perhaps we could add a pref change registrar to track the global preference

Reply:
i agree. can we do that in a follow up PR

Reply:
yes, sure